### PR TITLE
fix(annotations): use correct value for isOpen prop

### PR DESCRIPTION
**Pull request by ChenCodes**
Changes in this PR:
- [x] we are now using isEditing value to determine whether or not we should be showing the Cancel / Post buttons

The issue is that before when we clicked on the Modify button, the form input would show up but the Cancel / Post buttons would not show up until the user clicks to focus the input field.

The correct UX is that when the user clicks on the Modify button, we should immediately see the Cancel / Post buttons.
Screen Shot 2020-10-20 at 9 23 05 AM

### CODE REVIEW

File: src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js [hunk 1]
```diff
@@ -118,14 +119,14 @@ const AnnotationActivity = ({
                     <div>
                         <ActivityTimestamp date={createdAtTimestamp} />
                     </div>
-                    {isEditing && currentUser ? (
+                    {isEditing && isInputOpen && currentUser ? (
```
Comment (ChenCodes) [inline src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js:122]: If `isInputOpen` is false, then we shouldn't be showing the comment form.

File: src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js [hunk 2]
```diff
@@ -56,6 +56,7 @@ const AnnotationActivity = ({
 
     const handleEdit = () => {
         setIsEditing(true);
+        setIsInputOpen(true);
```
Comment (ChenCodes) [inline src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js:59]: Both `isEditing` and `isInputOpen` will be set to true if the user clicks on Modify button.

**Comment by booc0mtaco** [comment]
What are valid values for `isEditing` and `isInputOpen`? Can the UI get into a state where the former is true, and the latter is false, and vice versa?

File: src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js (hunk 1 above)
Comment (ConradJChan) [reply src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js:122]: Do we care to maintain the `isInputOpen` state? Would basing the logic off of `isEditing` alone be sufficient? Doesn't seem like the `onFocus` behavior is needed

**Comment by ChenCodes** [comment]
> What are valid values for `isEditing` and `isInputOpen`? Can the UI get into a state where the former is true, and the latter is false, and vice versa?

I realized that I didn't need to do both the isEditing / isInputOpen check. Just the `isEditing` check is enough because technically the form cannot be focused if the user hasn't clicked on the Modify button. If the user clicks on the Modify button, then both `isInputOpen` and `isEditing` will be true which means we should be showing the form.

The main fix in this change is that I was missing setting `isInputOpen` to be true in the edit click handler.

File: src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js (hunk 1 above)
Comment (ChenCodes) [reply src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js]: Yep I've addressed this in the latest commit - we don't need isInputOpen logic here.

**Comment by ConradJChan** [comment]
Can we get rid of `isInputOpen` state entirely? Do we have unit test coverage for the `CommentForm` being shown?

**Comment by ChenCodes** [comment]
> Can we get rid of `isInputOpen` state entirely? Do we have unit test coverage for the `CommentForm` being shown?

I don't think we can get rid of isInputOpen entirely because CommentForm has `onFocus` as a required property.

I'll be adding a unit test for CommentForm being shown

File: src/elements/content-sidebar/activity-feed/annotations/__tests__/AnnotationActivity.test.js [hunk 3]
```diff
@@ -95,6 +95,23 @@ describe('elements/content-sidebar/ActivityFeed/annotations/AnnotationActivity',
...
+                permissions: { can_edit: true },
+            },
+        };
+
+        const wrapper = mount(<AnnotationActivity {...mockActivity} {...activity} />);
+
+        wrapper.find('AnnotationActivityMenu').simulate('click');
+        wrapper.find('MenuItem').simulate('click');
+
+        expect(wrapper.exists('CommentForm')).toBe(true);
```
Comment (ConradJChan) [inline src/elements/content-sidebar/activity-feed/annotations/__tests__/AnnotationActivity.test.js:112]: Could we do a cancel button click and verify that the `CommentForm` doesn't exist as well?

**Review by ConradJChan — APPROVED** [review]

**Review by ConradJChan — APPROVED** [review]

**Pull request merged**